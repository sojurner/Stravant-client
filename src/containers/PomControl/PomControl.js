import React, { Component } from 'react';
import { connect } from 'react-redux';
import openSocket from 'socket.io-client';
import moment from 'moment';
// import { createActivity } from '../../helpers/apiCalls/apiCalls';
import { PomContainer } from '../../components/PomContainer/PomContainer';
import * as pomActions from '../../actions/pomAction';
import { getLastPomTime } from '../../helpers/helpers/helpers';

import './PomControl.css';

var socket = openSocket('http://localhost:5000');
export class PomControl extends Component {
  constructor() {
    super();
    this.state = {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      start: false,
      stop: true,
      save: false,
      hide: false,
      description: false,
      pomSummary: '',
      socketMessage: ''
    };
  }

  componentDidMount() {
    const { setPomHistory } = this.props;
    if (localStorage.getItem('pomHistory')) {
      const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
      setPomHistory(storageItem);
    }
  }

  togglePom = (event, string) => {
    const { start, hide } = this.state;
    if (start) {
      const toggleHide = !hide;
      this.setState({ hide: toggleHide });
    } else {
      this.sendSocketMsg(event);
      this.setState({ [string]: true, stop: false });
    }
    const { togglePomState, pomInfo } = this.props;
    togglePomState(!pomInfo.pomStatus);
  };

  sendSocketMsg = () => {
    const { firstName } = this.props.currentUser.info;
    socket.emit('pomStart', `${firstName} just started on Pom!`);
  };

  startMSecond = () => {
    let time = this.state.mSecond + 1;
    time < 10
      ? this.setState({ mSecond: time })
      : this.setState({ mSecond: 0 });
  };

  startSecond = () => {
    let time = this.state.second + 1;
    time < 60 ? this.setState({ second: time }) : this.setState({ second: 0 });
  };

  startMinute = () => {
    let time = this.state.minute + 1;
    time < 60 ? this.setState({ minute: time }) : this.setState({ minute: 0 });
  };

  startHour = () => {
    let time = this.state.hour + 1;
    this.setState({ hour: time });
  };

  stopTime = () => {
    const { firstName } = this.props.currentUser.info;
    const { pomHistory } = this.props.pomInfo;
    const timeKeys = Object.keys(pomHistory);
    const lastPom = timeKeys[timeKeys.length - 1];
    const currentTime = moment().format('llll');
    if (lastPom) {
      const difference = getLastPomTime(currentTime, lastPom);
      socket.emit(
        'pomEnd',
        `${firstName} finished the Pom!  Your last pom was ${difference}`
      );
    }
    this.setState({ start: false, stop: true });
  };

  addDescription = string => {
    this.setState({ description: string });
  };

  removeDescription = () => {
    this.setState({ description: '' });
  };

  resetTimer = () => {
    const { setPomHistory } = this.props;
    const { minute, second, hour } = this.state;
    const timeState = {
      second: second,
      minute: minute,
      hour: hour
    };
    if (localStorage.getItem('pomHistory')) {
      const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
      const newItem = {
        ...storageItem,
        [moment().format('llll')]: timeState
      };
      localStorage.setItem('pomHistory', JSON.stringify(newItem));
    } else {
      const itemToStorage = {
        [moment().format('llll')]: timeState
      };
      localStorage.setItem('pomHistory', JSON.stringify(itemToStorage));
    }
    const storageItem = JSON.parse(localStorage.getItem('pomHistory'));
    setPomHistory(storageItem);

    const timeSummary = `${second}s  ${minute}m  ${hour}h`;
    this.setState({
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      start: false,
      save: true,
      pomSummary: timeSummary
    });
  };

  removePom = (history, time) => {
    delete history[time];
    this.props.setPomHistory(history);
    localStorage.setItem('pomHistory', JSON.stringify(history));
  };

  render() {
    socket.on('pomClick', msg => {
      this.setState({ socketMessage: msg });
    });
    const { pomHistory } = this.props.pomInfo;
    // const styles = {
    //   pomStyle: {
    //     position: 'absolute',
    //     left: window.screenX + 10,
    //     top: window.screenY - 20
    //   },

    //   summaryStyle: {
    //     position: 'absolute',
    //     left: window.screenX + 25,
    //     top: window.screenY + 90
    //   },

    //   suggestStyle: {
    //     position: 'absolute',
    //     left: window.screenX + 25,
    //     top: window.screenY + 60
    //   },

    //   modalStyle: {
    //     position: 'absolute',
    //     left: window.screenX + 100,
    //     top: window.screenY - 20
    //   }
    // };
    const {
      mSecond,
      second,
      minute,
      hour,
      start,
      stop,
      description,
      hide,
      save,
      pomSummary,
      socketMessage
    } = this.state;
    return (
      <div className="pom-control">
        <h3 className="track-poms">Track Pomodoros</h3>
        <section className="pom-details">
          {socketMessage && <h4 className="socket-msg">{socketMessage}</h4>}
          <i
            class="fas fa-play-circle"
            onMouseEnter={() => this.addDescription('start')}
            onMouseLeave={this.removeDescription}
            onClick={event => {
              this.togglePom(event, 'start');
              if (!start && stop) {
                this.mSeconds = setInterval(this.startMSecond, 101);
                this.seconds = setInterval(this.startSecond, 1001);
                this.minutes = setInterval(this.startMinute, 60001);
                this.hours = setInterval(this.startHour, 3600001);
              }
            }}
          >
            {' '}
          </i>

          {(second > 0 || mSecond > 0 || hour > 0 || minute > 0) &&
            !hide &&
            start && (
              <i
                class="fas fa-stop-circle"
                onClick={() => {
                  this.stopTime();
                  this.removeDescription();
                  if (!stop) {
                    clearInterval(this.mSeconds);
                    clearInterval(this.seconds);
                    clearInterval(this.minutes);
                    clearInterval(this.hours);
                    this.resetTimer();
                  }
                }}
                onMouseEnter={() => this.addDescription('stop')}
                onMouseLeave={this.removeDescription}
              />
            )}

          {(second > 0 || mSecond > 0 || hour > 0 || minute > 0) &&
            !hide &&
            start && (
              <div className="time">
                <span className="hour">
                  {hour}
                  <div className="abbr">h </div>{' '}
                </span>
                <span className="minute">
                  {minute}
                  <div className="abbr">m </div>{' '}
                </span>
                <span className="second">
                  {second}
                  <div className="abbr">s </div>{' '}
                </span>
                <span className="mSecond">{mSecond}</span>
              </div>
            )}
          {!pomHistory && <p>You have no record Poms</p>}
          {description === 'start' &&
            !hide &&
            !start && <p className="pom-instruction">Start Pom?</p>}
          {description === 'start' &&
            !hide &&
            start && <p className="pom-instruction">Hide Pom?</p>}
          {description === 'stop' && (
            <p className="stop-instruction">End Pom?</p>
          )}

          {hide && <p className="stop-instruction">Show Pom?</p>}

          {save && (
            <h4>
              <div>Most recent:</div>
              ---
              {pomSummary}
              ---
            </h4>
          )}
        </section>

        <PomContainer removePom={this.removePom} pomHistory={pomHistory} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentUser: state.currentUser,
  pomInfo: state.pomInfo
});

const mapDispatchToProps = dispatch => ({
  setPomHistory: history => dispatch(pomActions.setPomHistory(history)),
  togglePomState: bool => dispatch(pomActions.togglePomState(bool))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PomControl);
