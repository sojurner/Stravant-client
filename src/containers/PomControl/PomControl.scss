.pom-instruction,
.stop-instruction {
  position: absolute;
  bottom: 35%;
  left: 47.8%;
  font-size: 1.2rem;
  font-family: daybold;
  color: azure;
}

.track-poms {
  font-family: daybold;
  font-size: 3rem;
  margin: 0.5rem 0 0 0;
  text-align: center;
  color: #d19e86;
}

img {
  cursor: pointer;
}
i {
  cursor: pointer;
}

.pom-details {
  text-align: center;
}

.socket-msg {
  position: absolute;
  left: %;
  font-size: 2rem;
  font-family: daybold;
}

.modal-timer {
  z-index: 1;
  background-color: #c2b8b5;
  padding: 1rem;
  text-align: center;
}

.fa-play-circle {
  color: #c2bcba;
  font-size: 4rem;
  margin: 6.5rem 2rem 1.5rem 2rem;
}

.fa-stop-circle {
  color: #d19e86;
  font-size: 4rem;
  margin: 1rem 2rem;
}

.pom-control {
  margin: auto;
  display: inline-block;
}

.fa-stop-circle:hover {
  color: rgb(131, 52, 52);
  text-align: right;
}

.modal {
  display: inline-block;
}

.fa-play-circle:hover {
  color: #979290;
}

.time {
  z-index: 1;
  background: rgb(177, 148, 148);
  border: 2px dotted black;
  position: absolute;
  padding: 3rem 1.5rem;
  left: 48%;
  margin: -12.5rem 0 0 -12.5rem;
  top: 40%;
  font-family: daybold;
  font-size: 4.5rem;
  animation: fadein 1s;
  border-radius: 8px;
  max-width: 25rem;
  min-width: 25rem;
}

.abbr {
  font-size: 2rem;
  display: inline-block;
}

@keyframes fadein {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

i {
  font-size: 2rem;
}
