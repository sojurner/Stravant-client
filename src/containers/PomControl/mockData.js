export const currentUser = {
  info: {
    accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
    userId: 34515396,
    firstName: 'Paul',
    gender: 'M',
    profilePic:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
  },
  totalStats: { Biked: 0, Ran: 6.472964574269732, Swam: 0 },
  weeklyStats: {}
};

export const pomInfo = {
  pomStatus: false,
  lastPom: '',
  pomHistory: {
    'Mon, Sep 10, 2018 9:12 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: true,
      showHistory: true,
      description: false
    },
    'Mon, Sep 10, 2018 11:43 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: false,
      showHistory: false,
      description: false
    },
    'Mon, Sep 10, 2018 11:58 PM': {
      mSecond: 0,
      second: 0,
      minute: 0,
      hour: 0,
      save: false,
      showHistory: false,
      description: false
    }
  }
};
