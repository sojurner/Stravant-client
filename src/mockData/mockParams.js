export const tokenParams = [
  'https://www.strava.com/oauth/token',
  {
    body:
      '{"client_id":28236,"client_secret":"829f12d9bdb8ce424edf87deff4551c117d3ed36","code":"r5423rffda4c234ctwtwert2ct"}',
    headers: { 'Content-Type': 'application/json' },
    method: 'POST'
  }
];

export const aggregateParams = [
  'https://www.strava.com/api/v3/athletes/28236/stats',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const weeklyParams = [
  'https://www.strava.com/api/v3/athlete/activities?before=1536123319&after=1536036919',
  {
    headers: {
      Authorization: 'Bearer a7eeea216a46a760a74c60acbffc3b55c66537c5',
      'Content-Type': 'application/json'
    },
    method: 'GET'
  }
];

export const userInfoParams = {
  token_type: 'Bearer',
  access_token: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  athlete: {
    id: 34515396,
    username: null,
    resource_state: 2,
    firstname: 'Paul',
    lastname: 'K',
    city: null,
    state: null,
    country: null,
    sex: 'M',
    premium: false,
    summit: false,
    created_at: '2018-09-04T22:53:48Z',
    updated_at: '2018-09-07T12:20:05Z',
    badge_type_id: 0,
    profile_medium:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    profile:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg',
    friend: null,
    follower: null,
    email: 'pykim0591@gmail.com'
  }
};

export const userStatsParams = {
  biggest_ride_distance: null,
  biggest_climb_elevation_gain: null,
  recent_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0,
    achievement_count: 0
  },
  recent_run_totals: {
    count: 5,
    distance: 10414.50015258789,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 9.439761638641357,
    achievement_count: 11
  },
  recent_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0,
    achievement_count: 0
  },
  ytd_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  ytd_run_totals: {
    count: 5,
    distance: 10415,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 10
  },
  ytd_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  all_ride_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  },
  all_run_totals: {
    count: 5,
    distance: 10415,
    moving_time: 3364,
    elapsed_time: 4832,
    elevation_gain: 10
  },
  all_swim_totals: {
    count: 0,
    distance: 0,
    moving_time: 0,
    elapsed_time: 0,
    elevation_gain: 0
  }
};

export const clubDataResultParam = [
  {
    id: 473825,
    resource_state: 2,
    name: 'Stravant Club',
    profile_medium:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/medium.jpg',
    profile:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246543/1/large.jpg',
    cover_photo:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/large.jpg',
    cover_photo_small:
      'https://dgalywyr863hv.cloudfront.net/pictures/clubs/473825/10246574/1/small.jpg',
    sport_type: 'running',
    city: 'Denver',
    state: 'Colorado',
    country: 'United States',
    private: false,
    member_count: 5,
    featured: false,
    verified: false,
    url: 'stravant'
  }
];

export const clubActivityDataParam = [
  {
    resource_state: 2,
    athlete: { resource_state: 2, firstname: 'Pamela', lastname: 'L.' },
    name: 'Morning Run',
    distance: 6871.2,
    moving_time: 2620,
    elapsed_time: 2821,
    total_elevation_gain: 17,
    type: 'Run',
    workout_type: null
  },
  {
    resource_state: 2,
    athlete: { resource_state: 2, firstname: 'Paul', lastname: 'K.' },
    name: 'Afternoon Run',
    distance: 501.6,
    moving_time: 149,
    elapsed_time: 183,
    total_elevation_gain: 0,
    type: 'Run',
    workout_type: 0
  }
];

export const weeklyDataParams = [
  [],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Afternoon Run',
      distance: 501.6,
      moving_time: 149,
      elapsed_time: 183,
      total_elevation_gain: 0,
      type: 'Run',
      workout_type: 0,
      id: 1832950216,
      external_id: '3D3F2956-3EC2-41F4-B4B8-5C0449E93948',
      upload_id: 1964982307,
      start_date: '2018-09-10T19:44:20Z',
      start_date_local: '2018-09-10T13:44:20Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.75, -105],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.75,
      start_longitude: -105,
      achievement_count: 1,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1832950216',
        summary_polyline: 'k{rqFndz_SbEeB~D|Fb@q@uApA',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.366,
      max_speed: 9.3,
      has_heartrate: false,
      elev_high: 1587,
      elev_low: 1585.6,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    },
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 2178.6,
      moving_time: 853,
      elapsed_time: 1064,
      total_elevation_gain: 0,
      type: 'Run',
      workout_type: 0,
      id: 1832102085,
      external_id: 'EE8C6C38-CD30-4D62-87BB-0CA540BA6A48',
      upload_id: 1964114698,
      start_date: '2018-09-10T12:31:39Z',
      start_date_local: '2018-09-10T06:31:39Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -104.99],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -104.99,
      achievement_count: 4,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1832102085',
        summary_polyline: 'wypqFnfx_SwGfDiJnQe_@ph@wEmDv@aCcDdGKgA',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 2.554,
      max_speed: 9.5,
      has_heartrate: false,
      elev_high: 1601.1,
      elev_low: 1585,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 3950,
      moving_time: 1069,
      elapsed_time: 1077,
      total_elevation_gain: 2.1,
      type: 'Run',
      workout_type: 0,
      id: 1827412827,
      external_id: 'DE69C2FB-C9EE-4FD8-ABCB-BDA8F89B50FD',
      upload_id: 1959242580,
      start_date: '2018-09-08T13:16:10Z',
      start_date_local: '2018-09-08T07:16:10Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -105.04],
      end_latlng: [39.74, -105.04],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -105.04,
      achievement_count: 3,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1827412827',
        summary_polyline:
          'apqqFr_b`SmH@{ElH{Ft\\wFaAeUjP~@tKbChGsAzMbBtFdXfAzGeBdHcIhAwDo@_IqEwMfDsItBk[',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.695,
      max_speed: 5.6,
      has_heartrate: false,
      elev_high: 1621.2,
      elev_low: 1618.4,
      pr_count: 3,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Lunch Run',
      distance: 1562,
      moving_time: 579,
      elapsed_time: 1086,
      total_elevation_gain: 4.7,
      type: 'Run',
      workout_type: 0,
      id: 1825796851,
      external_id: '3A54AFD1-4551-40FC-BDB3-A72100B29082',
      upload_id: 1957570524,
      start_date: '2018-09-07T17:49:18Z',
      start_date_local: '2018-09-07T11:49:18Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.75, -105],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.75,
      start_longitude: -105,
      achievement_count: 3,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1825796851',
        summary_polyline: 'sxrqFvez_SShBcDlB_DeGlAsCg@mDwFvAlE|ElBuD`EfFx@g@',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 2.698,
      max_speed: 6.1,
      has_heartrate: false,
      elev_high: 1586,
      elev_low: 1583,
      pr_count: 3,
      total_photo_count: 0,
      has_kudoed: false
    },
    {
      resource_state: 2,
      athlete: { id: 34515396, resource_state: 1 },
      name: 'Morning Run',
      distance: 2222.3,
      moving_time: 714,
      elapsed_time: 1422,
      total_elevation_gain: 2.6,
      type: 'Run',
      workout_type: 0,
      id: 1825117879,
      external_id: '3CAB0E99-AEBC-4B32-9FB6-96572B190C9B',
      upload_id: 1956872229,
      start_date: '2018-09-07T11:56:11Z',
      start_date_local: '2018-09-07T05:56:11Z',
      timezone: '(GMT-07:00) America/Denver',
      utc_offset: -21600,
      start_latlng: [39.74, -104.99],
      end_latlng: [39.75, -105],
      location_city: null,
      location_state: null,
      location_country: null,
      start_latitude: 39.74,
      start_longitude: -104.99,
      achievement_count: 0,
      kudos_count: 0,
      comment_count: 0,
      athlete_count: 1,
      photo_count: 0,
      map: {
        id: 'a1825117879',
        summary_polyline:
          'q}oqFx{w_SyMq@[nDcEr@}DrG}BwAi@vHkHV]dB|@lAoDhHwALMpDua@|Zi@sB{Ax@',
        resource_state: 2
      },
      trainer: false,
      commute: false,
      manual: false,
      private: false,
      visibility: 'everyone',
      flagged: false,
      gear_id: null,
      from_accepted_tag: false,
      average_speed: 3.112,
      max_speed: 6.6,
      has_heartrate: false,
      elev_high: 1607.8,
      elev_low: 1586,
      pr_count: 0,
      total_photo_count: 0,
      has_kudoed: false
    }
  ],
  [],
  []
];

export const clubActionParam = {
  clubName: 'Stravant Club',
  clubId: 473825,
  membersStats: {
    mostRecent: {
      name: 'Paul',
      distance: 1.4110627719080175,
      activity: 'Morning Run'
    },
    Paul: { totalDistance: 7.87, totalTime: 101, membersRecent: 'Morning Run' },
    Benjamin: {
      totalDistance: 1.17,
      totalTime: 20,
      membersRecent: 'Light rail trip'
    },
    Jesse: {
      totalDistance: 278.77,
      totalTime: 5297,
      membersRecent: 'Evening Run'
    },
    Pamela: {
      totalDistance: 62.88999999999999,
      totalTime: 635,
      membersRecent: 'Afternoon Run'
    }
  }
};

export const pomHistoryParam = {
  'Wed, Sep 12, 2018 11:45 AM': { second: 2, minute: 0, hour: 0 },
  'Wed, Sep 12, 2018 12:23 PM': { second: 1, minute: 0, hour: 0 }
};

export const accessTokenParam = {
  accessToken: 'a7eeea216a46a760a74c60acbffc3b55c66537c5',
  userId: 34515396,
  firstName: 'Paul',
  gender: 'M',
  profilePic:
    'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/APUIFaMcxPJyTx-ZbjWs8aCsKgkL96lV4w/mo/photo.jpg'
};

export const weeklyActionParam = {
  Wednesday: 2270.4,
  Monday: 2680.2,
  Saturday: 3950,
  Friday: 3784.3,
  Tuesday: 0,
  Thursday: 0,
  Sunday: 0
};

export const totalStatsParam = { Biked: 0, Ran: 7.883778744561839, Swam: 0 };
