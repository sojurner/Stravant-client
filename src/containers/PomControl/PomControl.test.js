import React from 'react';
import { PomControl } from './PomControl';
import * as mock from './mockData';
import { isDate } from 'moment';

describe('PomControl', () => {
  let wrapper;
  let state;
  let defaultState;
  let mockSetPom;
  let mockToggle;

  beforeEach(() => {
    localStorage.clear();
    mockSetPom = jest.fn();
    mockToggle = jest.fn();
    wrapper = shallow(
      <PomControl
        currentUser={mock.currentUser}
        pomInfo={mock.pomInfo}
        setPomHistory={mockSetPom}
        togglePomState={mockToggle}
      />
    );
  });

  defaultState = {
    description: false,
    hour: 0,
    mSecond: 0,
    minute: 0,
    pomSummary: '7s  2m  1h',
    save: true,
    second: 0,
    showHistory: false,
    start: false
  };

  state = {
    mSecond: 4,
    second: 7,
    minute: 2,
    hour: 1,
    save: false
  };

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should setState of save to false and togglePomState to have been called', () => {
    wrapper.instance().togglePom('start');

    expect(wrapper.state('start')).toEqual(true);

    expect(mockToggle).toHaveBeenCalled();
  });

  it('should increment state of mSecond by 1 and reset to 0 when it reaches 10', () => {
    wrapper.setState({ mSecond: 8 });

    wrapper.instance().startMSecond();

    expect(wrapper.state('mSecond')).toEqual(9);

    wrapper.instance().startMSecond();

    expect(wrapper.state('mSecond')).toEqual(0);
  });

  it('should increment state of second by 1 and reset to 0 when it reaches 60', () => {
    wrapper.setState({ second: 58 });

    wrapper.instance().startSecond();

    expect(wrapper.state('second')).toEqual(59);

    wrapper.instance().startSecond();

    expect(wrapper.state('second')).toEqual(0);
  });

  it('should increment state of Minute by 1 and reset to 1 when it reaches 60', () => {
    wrapper.setState({ minute: 58 });

    wrapper.instance().startMinute();

    expect(wrapper.state('minute')).toEqual(59);

    wrapper.instance().startMinute();

    expect(wrapper.state('minute')).toEqual(0);
  });

  it('should increment state of Hour by 1', () => {
    wrapper.setState({ hour: 0 });

    wrapper.instance().startHour();

    expect(wrapper.state('hour')).toEqual(1);

    wrapper.instance().startHour();

    expect(wrapper.state('hour')).toEqual(2);
  });

  it('should toggle state of description', () => {
    wrapper.instance().handleDescription();

    expect(wrapper.state('description')).toEqual(true);

    wrapper.instance().handleDescription();

    expect(wrapper.state('description')).toEqual(false);
  });

  it('should reset the state of times to true and save to 0', () => {
    wrapper.setState(state);

    wrapper.instance().resetTimer();

    expect(wrapper.state()).toEqual(defaultState);
  });

  it('should grab from local storage', () => {
    const local = {
      'Mon, Sep 10, 2018 9:12 PM': {
        mSecond: '0',
        second: '0',
        minute: '0',
        hour: '0',
        save: 'true',
        showHistory: 'true',
        description: 'false'
      }
    };
    const stringifiedLocal = JSON.stringify(local);
    let expectedLocal = localStorage.store;

    wrapper.setState(state);
    localStorage.setItem('pomHistory', stringifiedLocal);
    wrapper.instance().resetTimer();
    expect(mockSetPom).toHaveBeenCalled();
  });

  it('should set new item to storage if there is no pom history', () => {
    const expected = localStorage.store;

    wrapper.setState({ minute: 4, second: 5, hour: 10 });
    wrapper.instance().resetTimer();

    expect(localStorage.store).toEqual(expected);
  });

  it('should toggle state of history', () => {
    wrapper.setState({ showHistory: false });
    wrapper.instance().showPoms();

    expect(wrapper.state('showHistory')).toEqual(true);
  });

  it('should remove pom from store', () => {
    const local = {
      'Mon, Sep 10, 2018 9:12 PM': {
        second: '1',
        minute: '2',
        hour: '3'
      },
      'Mon, Sep 15, 2017 9:12 PM': {
        second: '4',
        minute: '3',
        hour: '1'
      }
    };
    const expected = {
      'Mon, Sep 15, 2017 9:12 PM': {
        second: '4',
        minute: '3',
        hour: '1'
      }
    };
    const stringifiedLocal = JSON.stringify(local);
    const stringifiedExpected = JSON.stringify(expected);

    localStorage.setItem('pomHistory', stringifiedLocal);

    wrapper.instance().removePom(local, 'Mon, Sep 10, 2018 9:12 PM');
    expect(localStorage.store).toEqual({ pomHistory: stringifiedExpected });
  });
  it('should match snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
