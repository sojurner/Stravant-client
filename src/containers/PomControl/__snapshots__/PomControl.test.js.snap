// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PomControl should match snapShot 1`] = `
<div
  className="pom-control"
>
  <h3
    className="track-poms"
  >
    Track Pomodoros
  </h3>
  <section
    className="pom-details"
  >
    <i
      class="fas fa-play-circle"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
       
    </i>
  </section>
  <PomContainer
    pomHistory={
      Object {
        "Mon, Sep 10, 2018 11:43 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": false,
          "second": 0,
          "showHistory": false,
        },
        "Mon, Sep 10, 2018 11:58 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": false,
          "second": 0,
          "showHistory": false,
        },
        "Mon, Sep 10, 2018 9:12 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": true,
          "second": 0,
          "showHistory": true,
        },
      }
    }
    removePom={[Function]}
  />
</div>
`;

exports[`PomControl should match snapshot 1`] = `
<div
  className="pom-control"
>
  <h3
    className="track-poms"
  >
    Track Pomodoros
  </h3>
  <section
    className="pom-details"
  >
    <i
      class="fas fa-play-circle"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
       
    </i>
  </section>
  <PomContainer
    pomHistory={
      Object {
        "Mon, Sep 10, 2018 11:43 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": false,
          "second": 0,
          "showHistory": false,
        },
        "Mon, Sep 10, 2018 11:58 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": false,
          "second": 0,
          "showHistory": false,
        },
        "Mon, Sep 10, 2018 9:12 PM": Object {
          "description": false,
          "hour": 0,
          "mSecond": 0,
          "minute": 0,
          "save": true,
          "second": 0,
          "showHistory": true,
        },
      }
    }
    removePom={[Function]}
  />
</div>
`;
